src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php

```diff
@@ -470,4 +470,70 @@ private function writeText(string $content, array $options = array())
             isset($options['raw_output']) ? !$options['raw_output'] : true
         );
     }
+
+    private function describeRouteMethods(array $methods)
+    {
+        if (!$methods) {
+            return '<info>ANY</info>';
+        }
+
+        $knownMethods = array(
+            'CONNECT' => true,
+            'DELETE' => true,
+            'GET' => true,
+            'HEAD' => true,
+            'OPTIONS' => true,
+            'PATCH' => true,
+            'POST' => true,
+            'PUT' => true,
+            'TRACE' => true,
+        );
+
+        $methodDescription = '';
+        foreach ($methods as $method) {
+            $methodDescription .= isset($knownMethods[$method]) ? $method : '<error>'.$method.'</error>';
+            $methodDescription .= '|';
+        }
+
+        return rtrim($methodDescription, '|');
+    }
+
+    private function describeRoutePath(string $path)
+    {
+        if ('' === $path || '/' === $path) {
+            return $path; // cheap short cuts
```

Comment:
We can remove the `cheap short cuts` comment because it's not needed to understand the code.